### PR TITLE
Add nyc taxi example for anomaly detection

### PR DESCRIPTION
**Pull request by zuhaagha**
Added anomaly detection example with nyc taxi dataset. Results are identical to NAB.

### CODE REVIEW

**Comment by zuhaagha** [comment]
@scottpurdy Please review. Not sure if there is this PR can be linked to an issue.

File: examples/opf/clients/nyctaxi/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# CPU
```
Comment (scottpurdy) [inline examples/opf/clients/nyctaxi/README.md]: This isn't CPU!

File: examples/opf/clients/nyctaxi/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# CPU 
+
+This custom client is similar to hotgym client and demonstrates how to configure an OPF client with a Numenta's Cortical Learning Algorithm (CLA) to detect anomalies. The internal model is constantly generating predictions about the future. The model is configured to output an "anomaly score", a value between 0 and 1 which measures the discrepancy between the actual input and its predicted value.
```
Comment (scottpurdy) [inline examples/opf/clients/nyctaxi/README.md]: CLA -> HTM

File: examples/opf/clients/nyctaxi/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+##Usage
+Simply run the nyctaxi_anomaly.py file to start the client.
+
+    ./nyctaxi_anomaly.py
+
+The parameters used to create the model are specified in `model_params.py`.
+
+##Output
+The output anomaly scores generated are written into `anomaly_scores.csv`. The ground truth labels for anomalies in this datset are known, including 5 major anomalies that occur during the NYC marathon, Thanksgiving, Christmas, New Years day, and a snow storm.
```
Comment (scottpurdy) [inline examples/opf/clients/nyctaxi/README.md]: s/datset/dataset/

File: examples/opf/clients/nyctaxi/README.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# CPU 
+
+This custom client is similar to hotgym client and demonstrates how to configure an OPF client with a Numenta's Cortical Learning Algorithm (CLA) to detect anomalies. The internal model is constantly generating predictions about the future. The model is configured to output an "anomaly score", a value between 0 and 1 which measures the discrepancy between the actual input and its predicted value.
+
+##Usage
```
Comment (scottpurdy) [inline examples/opf/clients/nyctaxi/README.md]: nit: put space after heading indicator

File: examples/opf/clients/nyctaxi/model_params.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2013, Numenta, Inc.  Unless you have an agreement
```
Comment (scottpurdy) [inline examples/opf/clients/nyctaxi/model_params.py]: year -> 2016

File: examples/opf/clients/nyctaxi/README.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+Simply run the nyctaxi_anomaly.py file to start the client.
+
+    ./nyctaxi_anomaly.py
+
+The parameters used to create the model are specified in `model_params.py`.
+
+##Output
```
Comment (scottpurdy) [inline examples/opf/clients/nyctaxi/README.md]: nit: space after heading indicator and newline after (before first line in the section)

**Comment by scottpurdy** [comment]
Remove the anomaly_scores.csv file from this. It gets generated by the code so we shouldn't include it in the repo.

File: examples/opf/clients/nyctaxi/README.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+Simply run the nyctaxi_anomaly.py file to start the client.
+
+    ./nyctaxi_anomaly.py
+
+The parameters used to create the model are specified in `model_params.py`.
+
+##Output 
+
+The output anomaly scores generated are written into `anomaly_scores.csv`. The ground truth labels for anomalies in this dataset are known, including 5 major anomalies that occur during the NYC marathon, Thanksgiving, Christmas, New Years day, and a snow storm.
```
Comment (scottpurdy) [inline examples/opf/clients/nyctaxi/README.md]: Can you add a list below this that specifies the exact timestamps (or row indices) of these events?

File: examples/opf/clients/nyctaxi/nyctaxi_anomaly.py [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+  "nupic.datafiles", "extra/nyctaxi/nyc_taxi.csv"
+)
+_OUTPUT_PATH = "anomaly_scores.csv"
+
+_ANOMALY_THRESHOLD = 0.9
+
+
+def getCSVData(dataPath):
+  try:
+    data = pandas.read_csv(dataPath)
```
Comment (scottpurdy) [inline examples/opf/clients/nyctaxi/nyctaxi_anomaly.py]: We don't have a dependency on pandas in NuPIC. Just use the built in `csv` package and manually parse the values.

Comment (scottpurdy) [inline examples/opf/clients/nyctaxi/nyctaxi_anomaly.py]: Or better yet, just hard code the min/max to the exact min/max in the data. Set these as constants in the file with a comment explaining that.

**Comment by scottpurdy** [comment]
Done with second round of review. I think that covers all of my feedback.

**Comment by scottpurdy** [comment]
👍  - is this ready for merge?

**Comment by zuhaagha** [comment]
Yes, ready now! :) Accidentally committed to this branch for my change on improving HTM scores but just reverted it.

**Pull request merged**